docs/errors.md

```diff
@@ -26,11 +26,102 @@ Note that this error also can occur for external import maps (those with `src=""
+
+## 6
+
+### Named AMD modules require the named-register extra
+
+SystemJS Error #6 occurs when you attempt to register a named AMD module, but have not included the [named-register extra](/README.md#extras).
+
+A named AMD module is created by calling `define('name', ...)`.
+
+To diagnose, first determine which module is the named AMD module. You can do so by checking the stacktrace in the browser console, or by placing a breakpoint in your devtools to pause on all uncaught exceptions.
+
+A common cause of this issue is that you are using webpack's [`output.library` option](https://webpack.js.org/configuration/output/#outputlibrary) or rollup's [`output.name` option](https://rollupjs.org/guide/en/#outputname), which both result in named AMD define. If your module is being imported by SystemJS, there is no benefit to keeping the library name, and you should remove it.
+
+Another common cause of this issue is for a third party scripts to call `define()` as part of their UMD build. Such scripts may call `define()` even when not triggered by `System.import()`, which can cause this error. In such cases, consider removing the SystemJS AMD extra entirely, so that they do not call `define()` at all.
+
+If none of those apply to your situation, consider adding the named-register extra:
+
+```html
+<script src="https://unpkg.com/systemjs/dist/system.js"></script>
+<script src="https://unpkg.com/systemjs/dist/extras/amd.js"></script>
+<script src="https://unpkg.com/systemjs/dist/extras/named-register.js"></script>
+```
+
+## 7
+
+### Failed to fetch module - wrong HTTP status
+
+SystemJS Error #7 occurs when SystemJS attempted to load a module with [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API), but the HTTP response status was not >= 200 and < 300.
+
+This error occurs when using either the module-types or transform extras. The module-types extension is included in `system.js`, but not in `s.js`.
+
+When using the transform extra, SystemJS uses `fetch()` to load all modules. However, when using the module-types extension (included in system.js), SystemJS uses `fetch()` instead of `<script>` to load modules whenever the [shouldFetch hook](/docs/hooks.md##shouldfetchurl---boolean) returns true. By default, this occurs only when the file extension indicates it is a [CSS, JSON, HTML, or WASM module](/docs/module-types.md).
+
+SystemJS checks the HTTP [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object to check if the HTTP response status is ["ok"](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok), and throws Error #7 if it is not ok.
+
+To diagnose the problem, identify which module failed to load. Then check the browser console and network tab of your devtools to find the HTTP status. In order for the module to successfully load, the status needs to be >= 200 and < 300.
+
+# SystemJS Warnings
+
+This sections lists the SystemJS warnings that you may encounter.
+
+## 51
```

Comment:
I think it's best not to introduce limits unless absolutely necessary. Since we are adding a new digit anyway, perhaps just prefix warnings with `W`? Could be done by just passing eg `'W1'` as a string into the errMsg function (despite polymorphism).